android/src/ti/bluetooth/TiBluetoothModule.java

```diff
@@ -8,301 +8,251 @@
+				Log.d(LCAT, "Found something " + device.getName());
+				if (device.getName() != null) {
+					Log.d(LCAT,
+							"Found: " + device.getName() + " "
+									+ device.getAddress());
+
+					ArrayList<String> ids = new ArrayList<String>();
+					for (ParcelUuid id : device.getUuids()) {
+						ids.add(id.toString());
+					}
+					KrollDict kd = new KrollDict();
+					kd.put("device", btDeviceProxy);
+
+					kd.put("name", device.getName());
+					kd.put("address", device.getAddress());
+					kd.put("bondState", device.getBondState());
+					kd.put("ids", ids.toArray());
+					fireEvent("didDiscoverPeripheral", kd);
+					BluetoothGatt bluetoothGatt = device.connectGatt(
+							appContext, false,
+							new BluetoothGattCallbackHandler(
+									TiBluetoothModule.this));
+					btScanner.stopScan(scanCallback);
+				}
+			}
+		}
+	};
+
+	private List<ScanFilter> scanFilters() {
+		String[] ids = {};
+		return scanFilters(ids);
+	}
+
+	private List<ScanFilter> scanFilters(String[] ids) {
+		List<ScanFilter> list = new ArrayList<ScanFilter>(1);
+		for (int i = 0; i < ids.length; i++) {
+			ScanFilter filter = new ScanFilter.Builder().setServiceUuid(
+					ParcelUuid.fromString(ids[i])).build();
+			list.add(filter);
+		}
+		return list;
+	}
+
+	@Kroll.method
+	public int getState() {
+		return currentState;
+	}
+
+	@Kroll.method
+	public boolean isScanning() {
+		return isScanning;
+	}
+
+	@Kroll.method
+	public void initialize() {
+		btManager = (BluetoothManager) appContext
+				.getSystemService(Context.BLUETOOTH_SERVICE);
+		btAdapter = btManager.getAdapter();
+		if (btAdapter != null) {
+			setCurrentState(btAdapter.getState());
+		} else {
+			setCurrentState(MANAGER_STATE_UNSUPPORTED);
+		}
+	}
+
+	@Kroll.method
+	public void startScan() {
+		this.startScan(scanmode);
+	}
+
+	@Kroll.method
+	public void startScan(int _scanmode) {
+		if (btAdapter != null) {
+			ScanSettings scanSettings = new ScanSettings.Builder().setScanMode(
+					_scanmode).build();
+			btScanner = btAdapter.getBluetoothLeScanner();
+			btScanner.startScan(scanFilters(), scanSettings, scanCallback);
+			// btScanner.startScan(scanCallback);
+			isScanning = true;
+		}
+	}
+
+	@Kroll.method
+	public void startScanWithServices(String[] ids) {
+		if (btAdapter != null) {
+			ScanSettings settings = new ScanSettings.Builder().setScanMode(
+					ScanSettings.SCAN_MODE_BALANCED).build();
+			btScanner = btAdapter.getBluetoothLeScanner();
+			btScanner.startScan(scanFilters(ids), settings, scanCallback);
+			isScanning = true;
+		}
+	}
+
+	// @Override
+	// public void eventListenerAdded(String eventName, int count, KrollProxy
+	// proxy) {
+	// super.eventListenerAdded(eventName, count, proxy);
+	//
+	// if (eventName.equals("didDiscoverPeripheral")) {
+	//
+	// }
+	// }
+
+	public void setCurrentState(int currentState) {
+		this.currentState = currentState;
+	}
+
+	@Kroll.method
+	public void setScanMode(int sm) {
+		scanmode = sm;
```

Comment:
See latest master: https://github.com/hansemannn/ti.bluetooth/blob/master/android/src/ti/bluetooth/TiBluetoothModule.java#L268
add a check so that the scanmode is not set to number which is not supported